deploy_heroku.py

```diff
@@ -0,0 +1,63 @@
+import argparse
+import dataclasses
+import json
+import os
+import subprocess
+import urllib.request
+from typing import List
+
+
+@dataclasses.dataclass
+class HerokuRelease:
+    id: str
+    version: int
+    status: str
+
+
+def get_latest_heroku_release(app: str, api_key: str) -> HerokuRelease:
+    request = urllib.request.Request(f"https://api.heroku.com/apps/{app}/releases")
+    for key, value in (
+            ("Accept", "application/vnd.heroku+json; version=3"),
+            ("Authorization", f"Bearer {api_key}"),
+            ("Content-Type", "application/json"),
+            ("Range", "version ..; order=desc,max=10;"),
+    ):
+        request.add_header(key, value)
+
+    with urllib.request.urlopen(request) as response:
+        assert response.getcode() == 200, response.getcode()
```

Comment:
Kommentar von @zyv aus #3 


Das ist sehr interessant - so habe ich angefangen, aber dann als ich es live (denke mal mit sepacetamol) ausprobiert habe kam in der tat 206, also ich habe ich geändert - kann es sein, dass wenn du weniger als 10 hast kommt 206? Dann kann man die Prüfung machen nach dem motto


```python
assert 200 <= response.getcode() < 300, response.getcode()
```

Reply:
@zyv 
Ich habe es jetzt getestet und in der Dokumentation noch einmal nachgelesen.
206 kommt zurück, wenn es mehr  Einträge, als `max` gibt.
Wir haben `max` auf 10 gesetzt, somit kam bei meiner TestApp `200` zurück, bei allen anderen, da schon viele Deployments gelaufen sind `206`.

Mein Vorschlag wäre folgender:

```suggestion
    assert response.getcode() in (200, 206), response.getcode()
```

Reply:
Ist auch für mich ok, mein Vorschlag deckt alle positive response codes falls die noch irgendwelche hacken haben, aber wenn du strickt sein willst, kann man so machen.

Reply:
https://devcenter.heroku.com/articles/platform-api-reference#successful-responses
Das sind die Response Codes, denke mit `200` und `206` werden wir auskommen.